torch/ao/quantization/fx/prepare.py

```diff
@@ -239,20 +239,91 @@ def is_pattern_dtype_config_supported_by_backend(
-                is_input_arg_dtype_supported_by_backend(
-                    arg, input_node, node_name_to_target_dtype, dtype_config, backend_config)
+        for arg in list(input_node.args) + list(input_node.kwargs.values()):
+            supported = supported and is_input_arg_dtype_supported_by_backend(
+                arg, input_node, node_name_to_target_dtype, dtype_config, backend_config)
         # check if output dtype is supported
         supported = supported and is_output_dtype_supported_by_backend(
             output_node, node_name_to_target_dtype, dtype_config)
         if supported:
-            return True
-    return False
+            return dtype_config
+    return None
+
+def _get_qconfig_compatible_with_dtype_config(
+        qconfig: QConfig,
+        dtype_config: DTypeConfig,
+        pattern: Pattern) -> QConfig:
+    """
+    Return an updated QConfig that satisfies constraints from the backend specified through `dtype_config`.
+
+    We require the QConfig quantization range to be a subset of the backend's quantization range,
+    and the QConfig min scale value to be equal to the backend's min scale value. If either the
+    quantization range or the min scale value is not specified in the QConfig, then we will borrow
+    the corresponding values from the backend.
+
+    If no changes to the QConfig are required, then the original QConfig is returned as is.
+    """
+    def get_new_activation_post_process(
+            activation_post_process: Union[_PartialWrapper, Type[ObserverBase]],
+            dtype_with_constraints: DTypeWithConstraints,
+            debug_string: str) -> Union[_PartialWrapper, Type[ObserverBase]]:
+        """
+        Return an updated activation post process that satisfies the constraints of the backend.
+        If the backend constraints cannot be satisfied, an error will be thrown instead.
+        If no changes to the activation post process are required, then the original activation
+        post process is returned as is.
+        """
+        extra_keywords = {}
+        backend_quant_min = dtype_with_constraints.quant_min
+        backend_quant_max = dtype_with_constraints.quant_max
+        backend_scale_min = dtype_with_constraints.scale_min
+        if isinstance(activation_post_process, _PartialWrapper):
+            qconfig_quant_min = activation_post_process.p.keywords.get("quant_min")
+            qconfig_quant_max = activation_post_process.p.keywords.get("quant_max")
+            # TODO: rename eps to scale_min
+            qconfig_scale_min = activation_post_process.p.keywords.get("eps")
+        else:
+            qconfig_quant_min, qconfig_quant_max, qconfig_scale_min = None, None, None
+        # quantization ranges
+        if backend_quant_min is not None and backend_quant_max is not None:
+            if qconfig_quant_min is None and qconfig_quant_max is None:
+                # If user did not specify quant range, take it from the backend
```

Comment:
I think @jerryzh168 raised a similar question above so I'm moving that discussion here. There are two options to handle the case when BackendConfig specifies a range but QConfig doesn't.

Option 1: QConfig defaults to BackendConfig's range, need to "update" user's QConfig (current approach in this PR)
Option 2: Require QConfig to also specify a range that falls within the backend's range

Option 2 will break a lot of existing use cases once we add the qmin/qmax to FBGEMM and QNNPACK BackendConfigs, because a lot of our [default QConfigs](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/qconfig.py#L239) don't specify qmin/qmax. Is that OK?

Reply:
If we throw an exception here as is, it would break almost all use cases that rely on `get_default_config`, because the observers there don't explicitly specify qmin/qmax. However, I think we can just check against the observer's default qmin/qmax instead, then throwing an exception here will be OK. I can make that change.

Reply:
I'd recommend changing the default qconfig, it would be good to minimize "magic" and just follow the behavior of the specified configuration.